modules/combined/test/tests/gap_heat_transfer_mortar/finite-2d-rz/finite.i

```diff
@@ -0,0 +1,356 @@
+    primary_variable = disp_x
+  []
+  [normal_x]
+    type = NormalMortarMechanicalContact
+    primary_boundary = plank_right
+    secondary_boundary = block_left
+    primary_subdomain = frictionless_primary_subdomain
+    secondary_subdomain = frictionless_secondary_subdomain
+    variable = frictionless_normal_lm
+    secondary_variable = disp_x
+    component = x
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+  []
+  [normal_y]
+    type = NormalMortarMechanicalContact
+    primary_boundary = plank_right
+    secondary_boundary = block_left
+    primary_subdomain = frictionless_primary_subdomain
+    secondary_subdomain = frictionless_secondary_subdomain
+    variable = frictionless_normal_lm
+    secondary_variable = disp_y
+    component = y
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+  []
+  [thermal_contact]
+    type = GapConductanceConstraint
+    variable = thermal_lm
+    secondary_variable = temp
+    k = 1
+    use_displaced_mesh = true
+    primary_boundary = plank_right
+    primary_subdomain = frictionless_primary_subdomain
+    secondary_boundary = block_left
+    secondary_subdomain = frictionless_secondary_subdomain
+    displacements = 'disp_x disp_y'
+  []
+[]
+
+[BCs]
+  [left_temp]
+    type = DirichletBC
+    variable = temp
+    boundary = 'plank_left'
+    value = 400
+  []
+
+  [right_temp]
+    type = DirichletBC
+    variable = temp
+    boundary = 'block_right'
+    value = 300
+  []
+
+  [left_x]
+    type = DirichletBC
+    variable = disp_x
+    boundary = plank_left
+    value = 0.0
+  []
+  [left_y]
+    type = DirichletBC
+    variable = disp_y
+    boundary = plank_bottom
+    value = 0.0
+  []
+  [right_x]
+    type = ADFunctionDirichletBC
+    variable = disp_x
+    boundary = block_right
+    function = '-0.04*sin(4*(t+1.5))+0.02'
+    preset = false
+  []
+  [right_y]
+    type = ADFunctionDirichletBC
+    variable = disp_y
+    boundary = block_right
+    function = '-t'
+    preset = false
+  []
+[]
+
+[Materials]
+  [plank]
+    type = ADComputeIsotropicElasticityTensor
+    block = 'plank'
+    poissons_ratio = 0.3
+    youngs_modulus = ${E_plank}
+  []
+  [block]
+    type = ADComputeIsotropicElasticityTensor
+    block = 'block'
+    poissons_ratio = 0.3
+    youngs_modulus = ${E_block}
+  []
+  [stress]
+    type = ADComputeFiniteStrainElasticStress
+    block = 'plank block'
+  []
+
+  [heat_plank]
+    type = ADHeatConductionMaterial
+    block = plank
+    thermal_conductivity = 2
+    specific_heat = 1
+  []
+  [heat_block]
+    type = ADHeatConductionMaterial
+    block = block
+    thermal_conductivity = 1
+    specific_heat = 1
+  []
+[]
+
+[Preconditioning]
+  [smp]
+    type = SMP
+    full = true
+  []
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = 'PJFNK'
+  petsc_options = '-snes_converged_reason -ksp_converged_reason'
+  petsc_options_iname = '-pc_type -mat_mffd_err -pc_factor_shift_type -pc_factor_shift_amount -snes_max_it'
```

Comment:
Are you reducing the default number of iterations with `-snes_max_it`? What's the reason?

Reply:
I like my problems to solve in a reasonable number of nonlinear iterations. If they don't, then I presume there's a problem with my algorithm

Reply:
Was not familiar with passing the flag instead of spelling it out through `nl_max_its`. I guess it's just the same.